src/ac/newformat.c

```diff
@@ -172,7 +172,11 @@ ASN1_SEQUENCE(AC) = {
 
 IMPLEMENT_ASN1_FUNCTIONS(AC)
 
-AC * AC_dup(AC *x) { return (AC*)ASN1_item_dup((&(AC_it)), x); }
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+AC * AC_dup(AC *x) { return ASN1_item_dup(ASN1_ITEM_rptr(AC), x); }
+#else
+AC * AC_dup(const AC *x) { return ASN1_item_dup(ASN1_ITEM_rptr(AC), x); }
```

Comment:
Hi Mischa, thanks for the patch. I can't have a deep look right now, but would you please try the following?
Since this is C, technically speaking the function taking a const pointer should suffice. From a quick test I get a warning about discarding the const qualifier when calling ASN_item_dup when using OpenSSL v. 1.1. To remove the warning I would do something like:
```
AC * AC_dup(const AC *x) {
  return ASN1_item_dup(
    ASN1_ITEM_rptr(AC),
#if OPENSSL_VERSION_NUMBER < 0x30000000L
    (AC*)
#endif
    x
  );
}
```
i.e. keeping only one prototype (the one with a const parameter) and adding a conditional cast in the implementation.

Reply:
Hi Francesco,
but that is not really ok, I would think?
In OpenSSL 1.1 `ASN1_item_dup()` is allowed to actually change its second argument, so casting away the `const` could then lead to a SEGV.

Reply:
Ah, ok, I didn't remember that. There is no man page for ASN1_item_dup and I didn't have time to look directly in the source code.

Reply:
Unfortunately it seems it doesn't have a manpage, but I'd say this is a generic point: since `ASN1_item_dup()` is defined with `void *` as second argument, and not as `const void *`, we would need to look inside the actual implementation(s) to see whether it is changing its argument and whether we could safely cast it to `AC *`. I had a quick look in the OpenSSL 1.1 source but couldn't figure it out easily. And even then, it could be that it at some point changed, so we'd need to check many of the implementations. In short: better not explicitly cast away a `const` (-;